### ISSUE TITLE
/v2/apps/<guid>/summary does't return newly set env var unless app is restarted.

### ISSUE THREAD

**Issue by stephanme** [cloudfoundry/cloud_controller_ng#1572]
## Issue

/v2/apps/\<guid>/summary does't return newly set env var unless app is restarted.

## Context

Looks like a regression in capi-release v1.91.0 / cf-deployment 12.33.0.
Worked in capi-release v1.89.0 / cf-deployment 12.25.0

## Steps to Reproduce

1. Deploy a testapp.  e.g. in a directory with an index.html file:

```
cf push testapp -m 128M -b binary_buildpack -c "python3 -m http.server 8080" --random-route
```

2. Set an env var

```
cf set-env testapp HELLO WORLD
```

3. Check that env var is set with /v2/apps/\<guid>/env endpoint

```
cf curl /v2/apps/$(cf app testapp --guid)/env
{
   ...
   "environment_json": {
      "HELLO": "WORLD"   <-- OK
   },
   "system_env_json": {
      "VCAP_SERVICES": {}
   },
   "application_env_json": {
      "VCAP_APPLICATION": {
         ...
         "application_name": "testapp",
         ...
      }
   }
}
```

4. Check that env var is set with /v2/apps/\<guid>/summary endpoint

```
cf curl /v2/apps/$(cf app testapp --guid)/summary
{
   "guid": "c2f01f96-8427-4d5e-bf00-1177249c1fe1",
   "name": "testapp",
   ...
   "buildpack": "binary_buildpack",
   "detected_buildpack": "binary",
   "detected_buildpack_guid": "27676514-5ef3-4789-8a15-a48cf2fb620b",
   "environment_json": {},  <-- FAILURE: should not be empty
   ...
}
```

5. Restart app with `cf restart testapp`

6. Check that env var is set with /v2/apps/\<guid>/summary endpoint

```
cf curl /v2/apps/$(cf app testapp --guid)/summary
{
   "guid": "c2f01f96-8427-4d5e-bf00-1177249c1fe1",
   "name": "testapp",
   ...
   "buildpack": "binary_buildpack",
   "detected_buildpack": "binary",
   "detected_buildpack_guid": "27676514-5ef3-4789-8a15-a48cf2fb620b",
   "environment_json": {
      "HELLO": "WORLD"    <-- OK
   },
   ...
}
```

## Expected result

environment_json in step 4 (summary endpoint) should contain HELLO var, i.e. show the same result as in step 6 after the restart or as in step 3 (env endpoint).

## Current result

environment_json in step 4 (summary endpoint) is empty, i.e. the set env vars are missing.

**Comment by boyan-velinov** [comment]
Hello,

We have observed something that can help you with the investigation. The issue cannot be reproduced on applications that were never started.

Here are steps to reproduce:

1. Push application without starting:
`cf push -f manifest.yml --no-start --random-route`

2. Set env var `foo`
`cf curl v2/apps/GUID -X PUT -d '{"environment_json":{"foo":"aaa"}}' `

3. Check that env var `foo` is present in `env` endpoint:
`cf curl v2/apps/GUID/env | grep foo` 

4. Check that env `foo` is present in `/summary` endpoint
`cf curl v2/apps/GUID/summary | grep foo`

5. Start the app
`cf start debug`

6. Set new env var `bar`
`cf curl v2/apps/GUID -X PUT -d '{"environment_json":{"bar":"bbb"}}' `

7. Check that env var `bar` is present in `env` endpoint:
`cf curl v2/apps/GUID/env | grep bar`

8. Check that env `bar` is NOT present in `summary` endpoint:
`cf curl v2/apps/GUID/summary | grep bar `

The weird part is step 4 – it seems that for an application that was pushed but never started this issue does not occur. It is different than step 8, where the app has been started at least one. Hope this could help.

Best Regards,
Boyan

**Comment by Gerg** [comment]
This issue was addressed by 45a9d11 and is available in [capi-release 1.94](https://github.com/cloudfoundry/capi-release/releases/tag/1.94.0).
